src/codegen/codegen_info.hpp

```diff
@@ -284,6 +284,31 @@ struct IndexSemantics {
         , size(size) {}
 };
 
+class LongitudinalDiffusionInfo {
```

Comment:
(optional) Some brief docstrings for the public functions would be useful.

Reply:
done